### PR TITLE
[WIP] US105680 Adding Lit poller

### PR DESCRIPTION
**Pull request by ChrisKraljevicD2L**
https://rally1.rallydev.com/#/289692574792d/detail/userstory/298380282124

### CODE REVIEW

**Comment by pbrown-d2l** [comment]
This is a good start, we will need this component in Lessons first, so it should be it's own new component instead of something inside d2l-sequence-viewer

**Comment by dbatiste** [comment]
Yeah I think this concept is something that is bound to come up in a few places.  I'm not convinced this should be a component though.  We should talk - I think @dlockhart is probably interested in this as well.

**Comment by dbatiste** [comment]
Is the only "behavior" that it starts polling when connected to the DOM, stops polling when it's disconnected, and dispatches an event (bubbling and composed?)?  Besides that, this is just `setInterval`?

**Comment by pbrown-d2l** [comment]
@dbatiste This is tied to a HPI for Energy Safety Canada. This is to help the student discover newly released content topics when release conditions are met (in there case SCORM topics being complete which are quite slow at this point). We need something to call our API the helps facilitate this. This is just to see if we are on the right track.

File: components/poller.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+		if (this.poll) {
+			const event = new CustomEvent('d2l-sequence-viewer-poll', {
+				message: 'youve been polled'
+			});
+			this.dispatchEvent(event);
+		}
+	}
+}
+
+customElements.define('poller', Poller);
```
Comment (dbatiste) [inline components/poller.js]: Note: custom element tag names are required to have a `-` in them, per web components specs.  Typically they are prefixed with an org abbreviation, in our case `d2l-`.

**Comment by dbatiste** [comment]
Ok.  I don't think I fully understand the usage in the SCORM context.  If this is going to be used by any of our HM components, then the simplest approach that would be almost the same code would be to use a `mixin` instead of a component, which would still allow it to hook into the HM component lifecycle.

File: components/poller.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+		super();
+
+		this.poll = false;
+		this.pollInterval = 10000;
+		this.intervalID = 0;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.intervalID = setInterval(this.pollEvent.bind(this), this.pollInterval);
```
Comment (dbatiste) [inline components/poller.js]: A minor thing - it looks like the `intervalID` can be "private", so it should be prefixed with an underscore. ex. `_intervalId`.

Note: properties that get updated typically result in `render()` being called.  The intervalID doesn't have any affect on the rendering of the component, so I'd be tempted to omit it from the `properties` object.  It's negligible in this case, but something to be aware of.

**Comment by ChrisKraljevicD2L** [comment]
@dbatiste a mixin is not feasible since this is required in [`smart-curriculum`](https://github.com/Brightspace/smart-curriculum) which is primarily React. I'll write this as an ES6 module instead and whatever consumes it can use it in their lifecycle hooks.

**Comment by dlockhart** [comment]
ES6 module sounds good. This seems kind of like a similar use case to the [dismissible stuff](https://github.com/BrightspaceUI/core/blob/master/helpers/dismissible.js) I just wrote, which was done as an ES6 module. It's up to the consumer to register/unregister for it in the appropriate places, but it meant that it could be used in web components and other places as well.

**Pull request closed without merging**